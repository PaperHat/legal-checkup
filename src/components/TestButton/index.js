import React from 'react'

export default function TestButton ({ handleClick }) {
  return <button onClick={handleClick}>Run test</button>
}
