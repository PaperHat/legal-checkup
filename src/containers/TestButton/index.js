import { connect } from 'react-redux'

import TestButton from '../../components/TestButton'
import { testCount } from '../../state'

function mapDispatchToProps (dispatch) {
  return {
    handleClick: () => dispatch(testCount())
  }
}

export default connect(undefined, mapDispatchToProps)(TestButton)
