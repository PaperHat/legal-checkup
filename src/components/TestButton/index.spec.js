import React from 'react'
import { shallow } from 'enzyme'

import TestButton from '.'

describe('components:TestButton', function () {
  it('renders the TestButton and CSS properties properly', () =>
    expect(
      toJson(shallow(<TestButton>TestButton</TestButton>))
    ).toMatchSnapshot())
})
