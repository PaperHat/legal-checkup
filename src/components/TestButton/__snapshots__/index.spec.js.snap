// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:TestButton renders the TestButton and CSS properties properly 1`] = `
<button>
  Run test
</button>
`;
